src/sys/quota.rs

```diff
@@ -1,78 +1,235 @@
-use {Errno, Result, NixPath};
+//! Set and configure disk quotas for users, groups, or projects.
+//!
+//! # Examples
+//!
+//! Enabling and setting a quota:
+//!
+//! ```rust,no_run
+//! # use nix::sys::quota::*;
+//! quotactl_on(QuotaType::USRQUOTA, "/dev/sda1", QuotaFmt::QFMT_VFS_V1, "aquota.user");
+//! let mut dqblk: Dqblk = Default::default();
+//! dqblk.set_blocks_hard_limit(10000);
+//! dqblk.set_blocks_soft_limit(8000);
+//! quotactl_set(QuotaType::USRQUOTA, "/dev/sda1", 50, &dqblk, QIF_BLIMITS);
+//! ```
+use std::default::Default;
+use std::{mem, ptr};
 use libc::{self, c_int, c_char};
+use {Errno, Result, NixPath};
 
-#[cfg(all(target_os = "linux",
-          any(target_arch = "x86",
-              target_arch = "x86_64",
-              target_arch = "arm")),
-          )]
-pub mod quota {
-    use libc::{self, c_int};
+struct QuotaCmd(QuotaSubCmd, QuotaType);
 
-    pub struct QuotaCmd(pub QuotaSubCmd, pub QuotaType);
-    pub type QuotaSubCmd = c_int;
+impl QuotaCmd {
+    fn as_int(&self) -> c_int {
+        (((self.0 as i32) << 8) | ((self.1 as i32) & 0x00ff)) as c_int
```

Comment:
I don't think this will work.  The `QuotaSubCmd` values are constants, defined by libc with values like 0x100, 0x200, etc.  So you should eliminate the shift.

Reply:
This was in there already, so I actually didn't take a look at it at all! But the implementation follows from the `QCMD()` macro in `/usr/include/linux/quota.h`, so I'm not certain what the problem is. on Linux `QSYNC` for example is a 24-bit value, we shift it up 8 bits and then or-in a 2 bit value at the bottom. That seems fine to me. What are you looking at that makes you think this is incorrect?

I actually think we should block until `QCMD` is added to `libc` and actually make this rely on that instead.

Reply:
Sorry, I was looking at the BSD definitions for quota commands, which are also similar to the Linux definitions when `_LINUX_QUOTA_VERSION < 2`